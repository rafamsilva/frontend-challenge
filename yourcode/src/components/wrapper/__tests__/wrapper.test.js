import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';

import Wrapper from '../';

describe('Wrapper Component', () => {
  let container;

  beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  describe('Render', () => {
    it('should render without crashing', () => {
      act(() => {
        render(<Wrapper />, container);
      });
    });
  });

  describe('Snapshot', () => {
    it('should match snapshot', () => {
      act(() => {
        render(<Wrapper />, container);
      });

      expect(pretty(container.innerHTML)).toMatchSnapshot();
    });
  });
});
