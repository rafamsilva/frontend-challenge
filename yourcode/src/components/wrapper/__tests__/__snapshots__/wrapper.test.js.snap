// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapper Component Snapshot should match snapshot 1`] = `"<div class=\\"sc-AxjAm icZVuM\\"></div>"`;
